gapis/api/vulkan/primeable_image_data.go

```diff
@@ -564,163 +565,219 @@ func (p *imagePrimer) newPrimeableImageData(img VkImage, opaqueBoundRanges []VkI
+				sizes := p.sb.levelSize(srcImgObj.Info().Extent(), srcImgObj.Info().Fmt(), level, aspect)
+				r := ipRenderRecipe{
+					inputAttachmentImage:  srcImg,
+					inputAttachmentAspect: VkImageAspectFlagBits_VK_IMAGE_ASPECT_COLOR_BIT,
+					renderImage:           dstImg,
+					renderAspect:          aspect,
+					layer:                 layer,
+					level:                 level,
+					renderRectX:           int32(0),
+					renderRectY:           int32(0),
+					renderRectWidth:       uint32(levelSize.width),
+					renderRectHeight:      uint32(levelSize.height),
+					wordIndex:             uint32(0),
+					framebufferWidth:      uint32(sizes.width),
+					framebufferHeight:     uint32(sizes.height),
+				}
+				recipes = append(recipes, r)
+			})
+
+		kits, err := kb.BuildRenderKits(p.sb, recipes...)
+		if err != nil {
+			return nil, log.Errf(p.sb.ctx, err, "failed to build render kits from recipes")
+		}
+		primeable.kits = kits
+
+		return primeable, nil
+	}
+
+	primeByImageStore := (!primeByCopy) && (!primeByRendering) && ((srcImgObj.Info().Usage() & storageBit) != 0)
+	if primeByImageStore {
+		queue := getQueueForPriming(p.sb, srcImgObj, VkQueueFlagBits_VK_QUEUE_COMPUTE_BIT)
+		if queue.IsNil() {
+			return nil, log.Errf(p.sb.ctx, nilQueueErr, "[Building primeable image data that can be primed by host data imageStore operation, image: %v]", srcImg)
+		}
+		if !GetState(p.sb.newState).Queues().Contains(queue.VulkanHandle()) {
+			return nil, fmt.Errorf("Queue: %v does not exist", queue)
+		}
+
+		dev := queue.Device()
+		primeable := &ipPrimeableStoreKits{img: srcImg, queue: queue.VulkanHandle(), kits: []ipStoreKit{}}
+
+		recipes := []ipStoreRecipe{}
+		walkImageSubresourceRange(p.sb, srcImgObj, p.sb.imageWholeSubresourceRange(srcImgObj),
+			func(aspect VkImageAspectFlagBits, layer, level uint32, levelSize byteSizeAndExtent) {
+				r := ipStoreRecipe{
+					inputImage:   srcImg,
+					inputAspect:  aspect,
+					outputImage:  dstImg,
+					outputAspect: aspect,
+					layer:        layer,
+					level:        level,
+					wordIndex:    uint32(0),
+					extentWidth:  uint32(levelSize.width),
+					extentHeight: uint32(levelSize.height),
+					extentDepth:  uint32(levelSize.depth),
+					offsetX:      int32(0),
+					offsetY:      int32(0),
+					offsetZ:      int32(0),
+				}
+				recipes = append(recipes, r)
+			})
+		kb := p.GetStoreKitBuilder(dev)
+		kits, err := kb.BuildStoreKits(p.sb, recipes...)
+		if err != nil {
+			return nil, log.Errf(p.sb.ctx, err, "failed at build store kits for priming storage image")
+		}
+		primeable.kits = kits
+		return primeable, nil
+	}
+	primeByPreinitialization := (!primeByCopy) && (!primeByRendering) && (!primeByImageStore) && (srcImgObj.Info().Tiling() == VkImageTiling_VK_IMAGE_TILING_LINEAR) && (srcImgObj.Info().InitialLayout() == VkImageLayout_VK_IMAGE_LAYOUT_PREINITIALIZED)
+	if primeByPreinitialization {
+		// TODO:
```

Comment:
We should probably return an error here, if we ever run into this case. 
That way we won't forget.